### ISSUE TITLE
Add a remark about watching objects/array refs

### ISSUE THREAD

**Issue by silentmantra** [vuejs/docs#2454]
From the docs at least for me it's not obvious that watching a ref is actually shallow watching it's value only.  
The problem that we know that an object/array ref make the object/array accessed through the ref's value reactive.
So intuitively it could seem that watching an object/array ref is deep, but it's not.

I think a remark about this case should be added in the docs, otherwise I see periodically that kind of questions on the stackoverflow:

https://stackoverflow.com/questions/76786334/cannot-watch-an-array-ref-in-vue-3/76786671

**Comment by alexchexes** [comment]
Absolutely. There is inconsistent terminology usage. The [deep watchers section](https://vuejs.org/guide/essentials/watchers.html#deep-watchers) states the following:

> When you call `watch()` directly **on a reactive object**, it will implicitly create a **deep watcher**.

At the same time, the [`ref()` documentation section](https://vuejs.org/api/reactivity-core.html#ref) states:

> `ref()`
> Takes an inner value and returns a **reactive and mutable ref object**.

But actually, when you call `watch()` on `ref` (which is a **reactive object** according to the `ref()` documentation), it will not create a deep watcher.

So this needs to be clarified, as the current phrasing is misleading.

**Comment by soletan** [comment]
I was searching the documentation for a hint on why [this example](https://play.vuejs.org/#eNqVkjFPwzAQhf/KyUtbNQpFwBLaSoA6wAAI2OoOUXpNUxLbsp22UuT/ztkhaQdUiS2+973Lu7Mb9qBUvK+RJWxqMl0oCwZtreZcFJWS2kIDGjcRHFKbbcHBRssKBuQYcMFFJoWxkG1TkaOBmUeHMIHR/UksLFadtGwyWQubwJ1bjTwSug5bJoISZnNouICuY7xPyxphPKZ+rm36H0cEzRpRJWB1jb/+YGyp5WRFubpI1xNHwPSq3QLNTwdiVZlapFP/hwSapp/YOW85w1jErKHBN0Ue74wUtNcQj7NMVqooUb8pW9BiOKNGXvFaWpby8BJqPmrU1bMtZt9/1Hfm6GucvWs0qPfIWa/ZVOdoW3nx+YpH+u7FSq7rkugL4gcaWdY+Y4s91mJNsc+4kPY5vI5C5F9mcbQoTDeUD+pJF3jO6Kk8XRj9FPcmvg0+ujjmfgCBm95m) does not work (showing "changes: 2"). Stumbling upon the same statement in docs about `watch()` deeply watching by default was definitely confusing here.
